web/template/partial/stream/chat/chat.gohtml

```diff
@@ -1,60 +1,61 @@
 {{define "chat"}}
     {{- /*gotype: TUM-Live/web.ChatData*/ -}}
     {{$stream := .IndexData.TUMLiveContext.Stream}}
+    {{$userName := .IndexData.TUMLiveContext.User.Name}}
+    {{$userId := .IndexData.TUMLiveContext.User.ID}}
     <div x-cloak
-         x-data="{orderByLikes: false, replyTo: 0, message: '', anonymous: false, messages: [], activePoll: null, pollResult: null, isPopUp: {{.IsPopUp}}, admin: {{.IsAdminOfCourse}}, showPollCreateUI: false, pollQuestion: '', pollOptions: [{ answer: 'Yes' }, { answer: 'No' }]}"
-         x-init="
-         messages = await watch.fetchMessages({{$stream.ID}});
-         fetch('/api/chat/{{$stream.ID}}/active-poll').then(res => res.json()).then(d => {activePoll=d});
-         $nextTick(() => { watch.scrollToBottom() });
-         "
-         x-on:chatmessage.window="e => {messages.push(e.detail)}"
-         x-on:chatreply.window="e => {messages.find(m=>m.ID===e.detail.replyTo.Int64).replies.push(e.detail);}"
-         x-on:chatlike.window="e => {messages.find(m=>m.ID===e.detail.likes).likes=e.detail.num; $dispatch('reorder')}"
-         x-on:chatdelete.window="e => {messages.find(m=>m.ID===e.detail.delete).deleted=true;}"
-         x-on:chatnewpoll.window="e => { if (!anonymous) { pollResult = null; activePoll = { ...e.detail, selected: null }; } }"
-         x-on:polloptionvotesupdate.window="e => { activePoll = { ...activePoll, pollOptions: activePoll.pollOptions.map(pollOption => pollOption.id === e.pollOptionId ? { ...pollOption, votes: e.votes } : pollOption) }; }"
-         x-on:polloptionresult.window="e => { activePoll = null; pollResult = e.detail; }"
-         x-on:chatresolve.window="e => {messages.find(m=>m.ID===e.detail.resolve).resolved=true;}"
-         x-on:chatapprove.window="messages = await watch.fetchMessages({{$stream.ID}}); $nextTick(() => { watch.scrollToBottom() })"
-         @reorder="watch.sortMessages(messages, orderByLikes)"
+         x-data="watch.initChat({{.IsAdminOfCourse}},{{$stream.ID}}, {{$userId}},'{{$userName}}');"
+         x-init="await c.loadMessages(); await c.poll.load(); $nextTick(() => { watch.scrollToBottom() })"
```

Comment:
Everything is so clean now !! 💯 
I would say you could consider here: `await Promise.all([c.loadMessages(), c.poll.load()]);`.